cli/cmd/start.go

```diff
@@ -48,45 +49,64 @@ func NewStartCmd() *cobra.Command {
 	return startCmd
 }
 
+// startWatchdog starts tarantool instance with watchdog.
+func startWatchdog(ttExecutable string, instance running.InstanceCtx) error {
+	appName := running.GetAppInstanceName(instance)
+	// If an instance is already running don't try to start it again.
+	// For restarting an instance use tt restart command.
+	procStatus := process_utils.ProcessStatus(instance.PIDFile)
+	if procStatus.Code == process_utils.ProcStateRunning.Code {
+		log.Infof("The instance %s (PID = %d) is already running.", appName, procStatus.PID)
+		return nil
+	}
+
+	log.Infof("Starting an instance [%s]...", appName)
+
+	newArgs := []string{"start", "--watchdog", appName}
+
+	wdCmd := exec.Command(ttExecutable, newArgs...)
+	return wdCmd.Start()
+}
+
+// startInstancesUnderWatchdog starts tarantool instances under tt watchdog.
+func startInstancesUnderWatchdog(instances []running.InstanceCtx) error {
+	ttBin, err := os.Executable()
+	if err != nil {
+		return err
+	}
+
+	for _, instance := range instances {
+		if err := startWatchdog(ttBin, instance); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
 // internalStartModule is a default start module.
 func internalStartModule(cmdCtx *cmdcontext.CmdCtx, args []string) error {
 	if !isConfigExist(cmdCtx) {
 		return errNoConfig
 	}
 
+	if cmdCtx.Cli.TarantoolCli.Executable == "" {
```

Comment:
There is no need for the start command to collect apps and instances info if there is no tarantool executable. If tarantool presents - we collect instances info, and perform other checks to make sure that we are able to start found instances. Yes, we can move a check for binary presence after `FillCtx` and this will work, but this is not optimal from my point of view.